modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -48,6 +49,21 @@ public static bool InheritsFrom(this ITypeSymbol? symbol, string assemblyName, s
             return false;
         }
 
+        public static string? GetDocumentationSummaryText(this ISymbol symbol)
+        {
+            var syntax = symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
+            while (syntax is VariableDeclaratorSyntax vds)
+            {
+                syntax = vds.Parent?.Parent;
+            }
+            var trivia = syntax?.GetLeadingTrivia();
```

Comment:
The initial approach was to use `GetDocumentationCommentXml`, but it requires setting `GenerateDocumentationFile=true`. For convenience, I changed it to the current method in the end.

Reply:
I don't think `GenerateDocumentationFile=true` should be required for that API to work, I'll make a quick MRP to confirm.

Reply:
It looks like `GetDocumentationCommentXml` doesn't work in Analyzers and Source Generators because of an upstream issue (https://github.com/dotnet/roslyn/issues/23673). So we'll probably have to retrieve from the trivia as a workaround like you were doing before.

Add a TODO comment so we can keep track of the upstream issue and potentially fix it in the future:

```csharp
// TODO: Can't use GetDocumentationCommentXml in source generators: https://github.com/dotnet/roslyn/issues/23673
```